platform/contracts/admin/src/lib.rs

```diff
@@ -32,31 +36,113 @@ pub fn instantiate(
     deps: DepsMut<'_>,
     _env: Env,
     _info: MessageInfo,
-    msg: InstantiateMsg,
+    InstantiateMsg::Instantiate {
+        contract_owner,
+        contracts,
+    }: InstantiateMsg,
 ) -> ContractResult<CwResponse> {
     versioning::initialize(deps.storage, CONTRACT_VERSION)?;
 
-    msg.validate(&deps.querier)?;
+    ContractOwnerAccess::new(&mut *deps.storage).grant_to(&contract_owner)?;
```

Comment:
[everywhere] we have agreed on using `deref_mut()`

Reply:
While yes, we did, I am not sure we are doing the correct thing, because this is a **_reborrow_** rather then de-referencing.

In any case, here I think I can use `DepsMut::branch` to borrow it instead of moving the ownership, which I suppose is how they intended it to be used, but I'd like you to please (re)consider it as I haven't actually seen other projects call it manually.

Reply:
offline we have agreed on branching when a mutable ref is needed and dereferencing otherwise